### PR TITLE
Add a snippet class for generating the Intercom snippet

### PR DESCRIPTION
**Pull request by mckenna**
This adds a `Snippet` class to the Node client, which is exposed through the client like so:

```javascript
const { Snippet } = require('intercom-client');
const settings = {
  verificationSecret: 'abc123',
  app_id: 'xyz789',
  user_id: 1,
  email: 'peter@intercom.io',
  name: 'Peter McKenna',
  company: {
    id: 123,
    name: 'Intercom'
  }
};
const snippet = new Snippet(settings);
res.send(snippet.create());
```
It returns the full snippet, including the `user_hash`.

### CODE REVIEW

File: lib/snippet.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        intercomSettings.push(`${key}: { ${this.settingsToString(settings[key])} }`);
+      } else {
+        if (typeof settings[key] === 'string') {
+          intercomSettings.push(`${key}: "${settings[key]}"`);
+        } else {
+          intercomSettings.push(`${key}: ${settings[key]}`);
+        }
+      }
+    });
+    return intercomSettings.join(', ');
```
Comment (irvinebroque) [inline lib/snippet.js]: `intercomSettings` needs to be an array?

Comment (mckenna) [reply lib/snippet.js]: Doesn't have to be. I figured keeping each key / value pair as an element in an array makes it much easier to join them into a string and not have to worry about where you need to add commas.

Comment (irvinebroque) [reply lib/snippet.js]: 👍

Comment (irvinebroque) [reply lib/snippet.js]: [JSON.stringify](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify) won't work here?

Comment (mckenna) [reply lib/snippet.js]: No, JSON.stringify will quote literally everything (including the braces for the hash), so you'd get this:

```javascript
const settings = { 
  app_id: 'abc123', 
  name: 'Peter McKenna', 
  user_id: 1 
};

// Returns "{"app_id":"abc123","name":"Peter McKenna","user_id":1}"
JSON.stringify(settings);
```

File: lib/snippet.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+export default class Snippet {
+  constructor(settings) {
+    if (!settings.app_id) {
+      throw new Error('You must provide an app_id in your Intercom settings');
+    }
+    if (!settings.verificationSecret) {
+      throw new Error('You must provide your verification secret in your Intercom settings');
+    }
+    if (!settings.user_id && !settings.email) {
```
Comment (kmiyashiro) [inline lib/snippet.js]: Can we make this snippet work for logged out mode as well?

Comment (mckenna) [reply lib/snippet.js]: Yep! I'll add that.

Comment (mckenna) [reply lib/snippet.js]: This supports logged out now.

**Pull request merged**